service/storage_service.cc

```diff
@@ -1433,6 +1433,17 @@ class topology_coordinator {
                     node = std::move(f).get();
                 }
 
+                if (_group0.is_member(node.id, true)) {
+                    // If we remove a node, we make it a non-voter early to improve availability in some situations.
+                    // There is no downside to it because the removed node is already dead.
```

Comment:
Let's be more precise -- it's not necessarily dead -- let's say something like "the removed node is already considered dead by us (otherwise we'd use `decommission`)"

It could be alive behind a network partition for example, but the rest of the cluster considers it dead and irrecoverable.